### PR TITLE
Update first-module.md

### PR DESCRIPTION
**Pull request by obwansan**
Asking for some points to be clarified or information to be added / stated explicitly to make the tutorial easier to follow. Thanks :-)

### CODE REVIEW

File: doc/book/in-depth-guide/first-module.md [hunk 1]
```diff
@@ -139,7 +139,7 @@ public function getConfig()
 }
 ```
 
-Reload your application and you'll see that everything remains as it was. Next
+Reload your application and you'll see that everything remains as it was [[This is confusing to me. If it means reload localhost:8080/blog, you get a 404 error as the route hasn't been properly configured yet. If it means reload localhost:8080, obviously everything will remain as it was as you haven't affected the landing page route. Might be my misunderstanding, but spelling out exactly what is meant would help. Also, would be good to keep phrasing consistent, e.g. "refresh your site" vs "reload your application"]]. Next
```
Comment (weierophinney) [inline doc/book/in-depth-guide/first-module.md]: Two comments:

- "refresh your site" has a very different connotation to "reload your application", as the latter can also have implications of "redesign and relaunch your site".

- Why would "reload your application" ever mean "reload localhost:8080/blog", when we have not even talked about defining that route yet?

The point we were trying to make with that statement was that creating, registering, and adding empty configuration for a new module _has no  visible effect on the application_.

I'm really not sure how to make that more clear; concrete suggestions would be more helpful here.

File: doc/book/in-depth-guide/first-module.md [hunk 2]
```diff
@@ -174,7 +174,7 @@ return [
 ];
 ```
 
-We've now created a route called `post` that listens to the URL
+We've now created a route called `blog` that listens to the URL
```
Comment (weierophinney) [inline doc/book/in-depth-guide/first-module.md:175]: Nice catch - thanks!

File: doc/book/in-depth-guide/first-module.md [hunk 3]
```diff
@@ -291,7 +291,7 @@ Message:
 Zend\View\Renderer\PhpRenderer::render: Unable to render template "blog/list/index"; resolver could not resolve to a file
 ```
 
-Now the application tells you that a view template-file can not be rendered,
+Now the application tells you that a view template-file cannot be rendered,
```
Comment (weierophinney) [inline doc/book/in-depth-guide/first-module.md:292]: `cannot` and `can not` are both acceptable forms; the main point is that we are consistent within the documentation.

I just grepped through this repository, and it looks like we have three cases where we use "can not" versus 10 where we use `cannot`.

I'll accept a patch that ensures we use exactly the same wording throughout.

File: doc/book/in-depth-guide/first-module.md [hunk 4]
```diff
@@ -304,9 +304,9 @@ file and add some dummy content to it:
 Before we continue let us quickly take a look at where we placed this file. Note
 that view files are found within the `/view` subdirectory, not `/src` as they
 are not PHP class files, but template files for rendering HTML. The path,
-however, deserves some explanation. First we have the lowercased namespace,
-followed by the lowercased controller name (without the suffix 'controller'),
-and lastly comes the name of the action that we are accessing (again without the
+however, deserves some explanation. First we have the lowercased namespace `blog`,
+followed by the lowercased controller name `list` (without the suffix 'controller'),
+and lastly comes the name of the action that we are accessing, `index` (again without the
```
Comment (weierophinney) [inline doc/book/in-depth-guide/first-module.md:307]: I like these changes! Thanks!

File: doc/book/in-depth-guide/first-module.md [hunk 5]
```diff
@@ -332,11 +332,11 @@ return [
 
 The above configuration tells the application that the folder
 `module/Blog/view/` has view files in it that match the above described default
-scheme. It is important to note that with this you can not only ship view files
+scheme [[Not clear to me what default scheme is being referred to here. Do you mean the path structure? Could it be stated explicitly?]] . It is important to note that with this you can not only ship view files
```
Comment (weierophinney) [inline doc/book/in-depth-guide/first-module.md]: We reference the "above described default scheme" — it's detailed two paragraphs prior as `view/{namespace}/{controller}/{action}.phtml`. In fact, it's the paragraph immediately prior to the cut in this diff!

I guess what I'm wondering is: should we NOT expect the reader to remember this information by this point? If this is the case, do you have any concrete suggestions that would help readers remember this fact?

File: doc/book/in-depth-guide/first-module.md [hunk 6]
```diff
@@ -332,11 +332,11 @@ return [
...
 for your module, but you can also overwrite view files from other modules.
 
 Reload your site now. Finally we are at a point where we see something different
-than an error being displayed! Congratulations, not only have you created a
+than an error being displayed! [[Would be good if it said here what you should see, e.g. the Skeleton Application page with Blog\ListController::indexAction() as the header.]] Congratulations, not only have you created a
```
Comment (weierophinney) [inline doc/book/in-depth-guide/first-module.md]: Noted; could you suggest wording for us to use, please?

**Review by weierophinney — CHANGES_REQUESTED** [review]
Thank you for the detailed feedback!

I've made notes on a few of your comments asking for clarifications and/or concrete suggestions. In a few places, I've pushed back a bit, as the information you're claiming isn't present... is? and I'm not sure how we could make it more clear.

Thanks in advance for your help in clarifying these points for other users!

File: doc/book/in-depth-guide/first-module.md (hunk 1 above)
Comment (obwansan) [reply doc/book/in-depth-guide/first-module.md]: Apologies, I think I got confused here with creating the /blog route that comes next (where reloading the page correctly generates a 404 error). That aside, I think it would be clearer to say "reload your application and you'll see that nothing changes". Might seem pedantic, but as the tutorial doesn't specify the previous state of the application my first thought was "how was it? / how should it have been?". 

I know the tutorial assumes you're using the 'album' skeleton app built in the Getting Started tutorial, but I created a new skeleton app from scratch for this tutorial. I think it would be even better to follow it up with your own statement: "creating, registering, and adding empty configuration for a new module has no visible effect on the application." (I have a tech writing background, so I guess I don't think you can ever spell things out too explicitly! : )

File: doc/book/in-depth-guide/first-module.md (hunk 2 above)
Comment (obwansan) [reply doc/book/in-depth-guide/first-module.md:175]: :-)

File: doc/book/in-depth-guide/first-module.md (hunk 3 above)
Comment (obwansan) [reply doc/book/in-depth-guide/first-module.md:292]: Cool. I agree, consistency is the most important thing.

File: doc/book/in-depth-guide/first-module.md (hunk 4 above)
Comment (obwansan) [reply doc/book/in-depth-guide/first-module.md:307]: :-)

File: doc/book/in-depth-guide/first-module.md (hunk 5 above)
Comment (obwansan) [reply doc/book/in-depth-guide/first-module.md]: I don't think it's so much a matter of the reader remembering as understanding what's being referred to. I worked it out, but I had to read it a few times (might just be me :-) I would rephrase it as:

"The above configuration tells the application that the folder ```module/Blog/view/``` has view files in it that match the standard path format: ```view/{namespace}/{controller}/{action}.phtml```.

[As a side note, I have just started my first developer role, and while I have a good grounding in PHP, frameworks and ZF are new to me. My overall impression of the tutorial is that in some places things are spelt out very clearly and in others a level of knowledge beyond framework/ZF newbie is assumed and makes it hard to follow. For example, the next sentence

"It is important to note that with this you can not only ship view files for your module, but you can also overwrite view files from other modules."

isn't at all clear to me. Maybe the tutorial is aimed at people with more PHP/coding/framework experience than I have, but I don't know what "ship view files for your module" means here, or why you would want to overwrite view files from other modules.].

File: doc/book/in-depth-guide/first-module.md (hunk 6 above)
Comment (obwansan) [reply doc/book/in-depth-guide/first-module.md]: "Finally we are at a point where we see something different than an error being displayed! You should see the standard ZF Skeleton Application template page with ```Blog\ListController::indexAction()``` as the ```<h1>``` header."

A screenshot would be good too :-)

File: doc/book/in-depth-guide/first-module.md (hunk 1 above)
Comment (weierophinney) [reply doc/book/in-depth-guide/first-module.md]: @obwansan The two changes you propose ("reload your application and you'll see that nothing changes", and adding the statement about "creating, registering...") make sense - can you update the patch to make those changes, please?

File: doc/book/in-depth-guide/first-module.md (hunk 5 above)
Comment (weierophinney) [reply doc/book/in-depth-guide/first-module.md]: I like the rephrasing! Let's go with that, please!

Regarding this:

> It is important to note that with this you can not only ship view files for your module, but you can also overwrite view files from other modules.

What this means is that, assuming another module uses the standard path format for view scripts, it's possible for you to provide substitutions for those view scripts within your own module by mimicing that structure, and providing configuration for it.

As an example, let's say we have another module, "News", and it has a controller "Updates" with the action "Breaking". Assuming the module follows the standard layout, it would have a view script in `view/news/updates/breaking.phtml`, and configuration like the following:

```php
'view_manager' => [
    'template_path_stack' => [
        'news' => __DIR__ . '/../view',
    ],
],
```

In our own `Blog` module, we could provide configuration for that alongside our own view script configuration:

```php
'view_manager' => [
    'template_path_stack' => [
        'blog' => __DIR__ . '/../view',
        'news' => __DIR__ . '/../view/news',
    ],
],
```

The above tells the view manager to look in our own `view/news/` subdirectory for matching templates.

If our `Blog` module is registered later than the `News` module, that means the view manager will scan the `Blog` module's `view/news/` subdirectory for matching templates _before_ scanning the `News` module — allowing us to selectively override view scripts.

I'm not sure if this is too much information to introduce at this point in the tutorial, though. Thoughts?

File: doc/book/in-depth-guide/first-module.md (hunk 6 above)
Comment (weierophinney) [reply doc/book/in-depth-guide/first-module.md]: Rewording sounds good!

I can supply the screenshot when it's time to merge.

File: doc/book/in-depth-guide/first-module.md (hunk 1 above)
Comment (obwansan) [reply doc/book/in-depth-guide/first-module.md]: Done : )

File: doc/book/in-depth-guide/first-module.md (hunk 5 above)
Comment (obwansan) [reply doc/book/in-depth-guide/first-module.md]: Updated the patch with the rephrasing.

Thanks for this detailed explanation. It's very clear, although I haven't learnt enough about ZF yet to fully understand how this would work. I think inserting this info would break the flow of the tutorial so I would keep the note as it is at the moment, with a slight clarification (if I understand it correctly):

"It is important to note that the view_manager configuration not only allows you to ship view files for your module, but also to overwrite view files from other modules."

and make the last six words a link to a separate topic on the subject with the same name. Maybe put it in a tutorial Appendix that could hold more detailed explanations of things that are important, but would break the flow of the tutorial?

File: doc/book/in-depth-guide/first-module.md (hunk 6 above)
Comment (obwansan) [reply doc/book/in-depth-guide/first-module.md]: Great : )

**Comment by weierophinney** [comment]
Thanks, @obwansan!
